### DISCUSSION TITLE
How to access the child Nodes from modelNode?

### DISCUSSION THREAD

**Discussion by HDROHIT** [sceneview/sceneview#155] [Q&A]
I'm trying to access the child nodes from the model node but its returning empty always. I have created the model and loaded the glb file on the model node. However we are not able to get the child elements after the glb file is loaded. Its always returning empty.

Please suggest. I have attached the code below.

`
Code:

class MainFragment : Fragment(R.layout.fragment_main) {

    lateinit var sceneView: SceneView
    lateinit var loadingView: View

    var isLoading = false
        set(value) {
            field = value
            loadingView.isGone = !value
        }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        sceneView = view.findViewById(R.id.sceneView)
        loadingView = view.findViewById(R.id.loadingView)

        isLoading = true

        val modelNode = ModelNode(
            position = Position(z = -4.0f),
            rotation = Rotation(x = 35.0f)
        )
        sceneView.addChild(modelNode)

        sceneView.cameraNode.transform = lookAt(
            eye = modelNode.worldPosition.let {
                Position(x = it.x, y = it.y + 0.5f, z = it.z + 2.0f)
            },
            target = modelNode.worldPosition,
            up = Direction(y = 1.0f)
        )

        lifecycleScope.launchWhenCreated {
            sceneView.environment = HDRLoader.loadEnvironment(
                context = requireContext(),
                lifecycle = lifecycle,
                hdrFileLocation = "environments/studio_small_09_2k.hdr",
                specularFilter = false
            )?.apply {
                indirectLight?.intensity = 50_000f
            }

            modelNode.loadModel(
                context = requireContext(),
                lifecycle = lifecycle,
                glbFileLocation = "models/spiderbot.glb",
                autoAnimate = true,
                scaleToUnits = 1.0f,
                centerOrigin = Position(x = 0.0f, y = 0.0f, z = 0.0f)
            )

            delay(1500)
            isLoading = false

            sceneView.cameraNode.smooth(
                lookAt(
                    eye = modelNode.worldPosition.let {
                        Position(x = it.x - 0.4f, y = it.y + 0.4f, z = it.z - 0.6f)
                    },
                    target = modelNode.worldPosition,
                    up = Direction(y = 1.0f)
                ),
                speed = 1.0f
            )
        }
    }
}

---`

**Comment by ThomasGorisse** [comment]
Hi.
Sorry about that but this feature will only be introduced within SceneView 1.0.0: https://gist.github.com/ThomasGorisse/4f9981047a8a3bf8002eaf58f4026475
We are working hard on it but it's not yet available.
